crates/iota/src/keytool.rs

```diff
@@ -589,8 +589,15 @@ impl KeyToolCommand {
             } => match IotaKeyPair::decode(&input_string) {
                 Ok(ikp) => {
                     info!("Importing Bech32 encoded private key to keystore");
-                    let key = Key::from(&ikp);
-                    keystore.add_key(alias, ikp)?;
+                    let mut key = Key::from(&ikp);
+                    keystore.add_key(alias.clone(), ikp)?;
+
+                    let alias = match alias {
+                        Some(x) => x,
+                        None => keystore.get_alias_by_address(&key.iota_address)?,
```

Comment:
Wait, so if no alias was passed in, this will always fail?

Reply:
No, a random one is generated

Reply:
Ah, I see. This is very weird. `add_key` should really return the added alias 😆 So then this check actually cannot possibly fail. How do we feel about making this slightly more explicit by calling `keystore.create_alias` first to resolve the alias and removing this match?

Reply:
But then `create_alias` is called twice. Do you think the match is that bad?

Reply:
Actually, I could just simply just call `get_alias_by_address` and not match at all